### PR TITLE
feat(orchestrator): workflow instance result page

### PR DESCRIPTION
**Pull request by jkilzi**
Addresses:
https://issues.redhat.com/browse/FLPATH-691
https://issues.redhat.com/browse/FLPATH-843
https://issues.redhat.com/browse/FLPATH-845

Demo:
[Screencast from 2024-01-12 18-37-09.webm](https://github.com/caponetto/backstage-plugins/assets/77008341/7c40502c-76e2-49fa-a92b-b1e7e87e805a)

### CODE REVIEW

**Comment by jkilzi** [comment]
The servicenow plug-in is failing...
https://github.com/caponetto/backstage-plugins/actions/runs/7511161007/job/20450424069?pr=73#step:3:3241
https://github.com/caponetto/backstage-plugins/actions/runs/7511161007/job/20450424139?pr=73#step:3:5366

File: .gitignore [hunk 1]
```diff
@@ -64,4 +64,4 @@ site
 # build cache
 .webpack-cache
 
-.idea/
```
[the remark this answers is not in the document]
Comment (jkilzi) [reply .gitignore:67]: Yes. it is duplicating line 61.
image

I found it useful to setup the .tmp dir inside the repo in order to mount the container volume there and avoid weird permission issues.

[the remark this answers is not in the document]
Comment (caponetto) [reply .gitignore:67]: Good catch!

File: plugins/orchestrator/src/components/next/WorkflowInstancePage.tsx [hunk 2]
```diff
@@ -28,6 +35,31 @@ export const WorkflowInstancePage = ({
 
   const isReady = React.useMemo(() => !loading && !error, [loading, error]);
 
+  const handleAbort = React.useCallback(async () => {
+    if (value) {
+      // TODO(jkilzi): Show a dialog instead for confirming the abort action of a running WF
```
[the remark this answers is not in the document]
Comment (jkilzi) [reply plugins/orchestrator/src/components/next/WorkflowInstancePage.tsx:40]: Yes, @yzhao583 was supposed to open a task for this on Friday.

File: plugins/orchestrator/package.json [hunk 3]
```diff
@@ -65,6 +65,8 @@
     "@material-ui/icons": "^4.11.3",
     "@material-ui/lab": "^4.0.0-alpha.45",
     "@monaco-editor/react": "^4.6.0",
+    "@mui/icons-material": "^5.15.3",
+    "@rjsf/core": "^3.2.1",
```
[the remark this answers is not in the document]
Comment (jkilzi) [reply plugins/orchestrator/package.json]: No, apparently it reappeared after my rebase. Removing it.
Addressed in cf30751742b38a6aa310fc9fc09f0d76dd6ee19b

File: plugins/orchestrator/src/components/next/WorkflowInstancePage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkilzi) [reply plugins/orchestrator/src/components/next/WorkflowInstancePage.tsx]: Addressed in f04797dbf5060932e00a6a57c276f39d309bb27a

File: plugins/orchestrator/src/components/next/WorkflowInstancePageContent.tsx [hunk 4]
```diff
@@ -118,37 +165,54 @@ export const WorkflowInstancePageContent = ({
...
-            divider={false}
-            className={styles.card}
-          >
+        <Grid item xs={6}>
+          <InfoCard title="Workflow definition" className={styles.card}>
             <WorkflowEditor
-              workflowId={processInstance.description}
-              kind={EditorViewKind.EXTENDED_DIAGRAM_VIEWER}
+              workflowId={processInstance.processId}
+              kind={EditorViewKind.AUTHORING}
```
[the remark this answers is not in the document]
Comment (jkilzi) [reply plugins/orchestrator/src/components/next/WorkflowInstancePageContent.tsx]: Addressed in 6008c743349acaffa1d2e95a402027c9dd0ef5e7

File: plugins/orchestrator/src/utils/NodeInstanceUtils.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import moment from 'moment';
+
+import { NodeInstance } from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { isNonNullable } from './TypeGuards';
+
+export const compareNodes = (nodeA: NodeInstance, nodeB: NodeInstance) => {
```
[the remark this answers is not in the document]
Comment (jkilzi) [reply plugins/orchestrator/src/utils/NodeInstanceUtils.ts:7]: Addressed in 61d7cc215c84348b1ab6f83a2e21e08150d40b59

**Review by caponetto — APPROVED** [review]

**Pull request merged**